sodium/src/secbuf.rs

```diff
@@ -232,16 +232,48 @@ impl SecBuf {
     /// helper for writing data to our internal buffer
     pub fn write(&mut self, offset: usize, data: &[u8]) -> Result<(), SodiumError> {
         if offset + data.len() > self.len() {
-            return Err(SodiumError::new("bad write offset / length"));
+            return Err(SodiumError::new("SecBuf write overflow"));
         }
-
         unsafe {
             let mut b = self.write_lock();
             std::ptr::copy(data.as_ptr(), (**b).as_mut_ptr().add(offset), data.len());
         }
+        Ok(())
+    }
+
+    // -- Helpers -- //
+
+    /// Check if the buffer is empty i.e. [0,0,0,0,0,0,0,0]
+    pub fn is_empty(&mut self) -> bool {
+        let buf = self.read_lock();
+        for i in 0..buf.len() {
+            if buf[i] != 0 {
+                return false;
+            }
+        }
+        true
+    }
 
+    /// Load the [u8] into the SecBuf
+    pub fn from_array(&mut self, data: &[u8]) -> Result<(), SodiumError> {
+        if (data.len() != self.len()) {
+            return Err(SodiumError::Generic(
+                "Input does not have same size as SecBuf".to_string(),
+            ));
+        }
+        let mut buf = self.write_lock();
+        for x in 0..data.len() {
+            buf[x] = data[x];
+        }
         Ok(())
     }
+
+    /// Return true if both buffers hold the same data
+    pub fn is_same(&mut self, other: &mut Self) -> bool {
+        let a = self.read_lock();
+        let b = other.read_lock();
+        **a == **b
```

Comment:
Feel like this should be implemented using PartialEq

Reply:
Tried but PartialEq doesn't work with `&mut` args :/

Reply:
huh... in that case, maybe we could make use of https://download.libsodium.org/doc/helpers#comparing-large-numbers ? because we have need of that as well.

Reply:
Just discovered that sodium_compare() was already implemented in util.rs
I reworked util.rs to secbuf_util.rs and maybe methods instead.
Moved all helpers there.